imsim/sky_model.py

```diff
@@ -0,0 +1,113 @@
+            Exposure time in seconds.
+        mjd : `float`
+            MJD of observation.
+        bandpass : `galsim.Bandpass`
+            Bandpass to use for flux calculation.
+        eff_area : `float`
+            Collecting area of telescope in cm^2. Default: Rubin value from
+            https://confluence.lsstcorp.org/display/LKB/LSST+Key+Numbers
+        logger : `logging.Logger`
+            Logger object.
+        """
+        self.exptime = exptime
+        self.mjd = mjd
+        self.eff_area = eff_area
+        self.bandpass = bandpass
+        self.logger = logger
+        self._rubin_sim_sky_model = skybrightness.SkyModel()
+
+    def get_sky_level(self, ra, dec):
+        """
+        Return the sky level in units of photons/arcsec^2 at the
+        specified coordinates.
+
+        Parameters
+        ----------
+        ra : `float`
+            Right Ascension in degrees.
+        dec : `float`
+            Declination in degrees.
+
+        Returns
+        -------
+        `float` : sky level in photons/arcsec^2
+        """
+        # Make a copy of the skybrightness.SkyModel object to avoid
+        # collisions with other threads running this code.
+        rubin_sim_sky_model = copy.deepcopy(self._rubin_sim_sky_model)
+
+        # Set the ra, dec, mjd for the sky SED calculation
+        with warnings.catch_warnings():
+            # Silence astropy IERS warnings.
+            warnings.simplefilter('ignore')
+            rubin_sim_sky_model.setRaDecMjd(ra, dec, self.mjd, degrees=True)
+
+        # Compute the flux in units of photons/cm^2/s/arcsec^2
+        wave, spec = rubin_sim_sky_model.returnWaveSpec()
+        lut = galsim.LookupTable(wave, spec[0])
+        sed = galsim.SED(lut, wave_type='nm', flux_type='flambda')
+        flux = sed.calculateFlux(self.bandpass)
+
+        # Return photons/arcsec^2
+        value = flux * self.eff_area * self.exptime
+
+        if self.logger is not None:
+            self.logger.info("Setting sky level to %.2f photons/arcsec^2 "
+                             "at (ra, dec) = (%s, %s)", value, ra, dec)
+        return value
+
+
+class SkyModelLoader(InputLoader):
+    """
+    Class to load a SkyModel object.
+    """
+    def getKwargs(self, config, base, logger):
+        req = {'exptime': float,
+               'mjd': float}
+        opt = {'eff_area': float}
+        kwargs, safe = galsim.config.GetAllParams(config, base, req=req, opt=opt)
+        kwargs['bandpass'] = base['bandpass']
+        kwargs['logger'] = galsim.config.GetLoggerProxy(logger)
+        return kwargs, safe
+
+
+def sky_level(config, base, value_type):
+    """
+    Use the rubin_sim skybrightness model to return the sky level in
+    photons/arcsec^2 at the center of the current WCS.
+    """
+    sky_model = galsim.config.GetInputObj('sky_model', config, base, 'sky_level')
+
+    kwargs, safe = galsim.config.GetAllParams(config, base)
+
+    wcs = galsim.config.BuildWCS(base['image'], 'wcs', base)
```

Comment:
I think `wcs = base['wcs']` should work here.  At the time when we calculate the sky level, the wcs should already be built and saved.  No need to remake it here.
Edit: given the below comment, I don't think you even need the wcs here.